src/compas_timber/ghpython/components/CT_Model/code.py

```diff
@@ -112,27 +112,37 @@ def get_joints_from_rules(self, beams, rules, topologies):
                         )
         return joints
 
-    def RunScript(self, Beams, JointRules, Features, MaxDistance, CreateGeometry):
-        if not Beams:
+    def RunScript(self, Elements, JointRules, Features, MaxDistance, CreateGeometry):
+        if not Elements:
             self.AddRuntimeMessage(Warning, "Input parameter Beams failed to collect data")
         if not JointRules:
             self.AddRuntimeMessage(Warning, "Input parameter JointRules failed to collect data")
-        if not (Beams):  # shows beams even if no joints are found
+        if not (Elements):  # shows beams even if no joints are found
             return
         if MaxDistance is None:
             MaxDistance = TOL.ABSOLUTE  # compared to calculted distance, so shouldn't be just 0.0
 
         Model = TimberModel()
         debug_info = DebugInfomation()
-        for beam in Beams:
-            # prepare beams for downstream processing
-            beam.remove_features()
+        beams = []
+        plates = []
+        for element in Elements:
+            if element is None:
+                continue
+            element.remove_features()
+            element.debug_info = []
+            Model.add_element(element)
+            if element.__class__.__name__ == "Plate":
+                plates.append(element)
+            elif element.__class__.__name__ == "Beam":
+                beams.append(element)
```

Comment:
I'd rather do:
`if isinstance(element, Plate)`

or better would be:
if element.is_plate

or even better than that would be:
we override `Model.add_element` and check it there

Reply:
```python3
    def add_element(self, element, **kwargs):
        node = super(TimberModel, self).add_element(element, **kwargs)

        if isinstance(element, Beam):
            self._beams.append(element)
        elif isinstance(element, Wall):
            self._walls.append(element)
        elif isinstance(element, Plate):
            self._plates.append(element)
        else:
            raise NotImplementedError("Element type not supported: {}".format(type(element)))
        return node
```